Online/Entity/OnlinePhysicalObject.cs

```diff
@@ -305,9 +308,28 @@ public static void HitByWeapon(OnlinePhysicalObject objectHit, OnlinePhysicalObj
             objectHit?.apo.realizedObject.HitByWeapon(weapon.apo.realizedObject as Weapon);
         }
         [RPCMethod]
-        public static void HitByExplosion(OnlinePhysicalObject objectHit, float hitfac)
+        public static void HitByExplosion(OnlinePhysicalObject objectHit, OnlinePhysicalObject sourceObject, Vector2 pos, int lifeTime, float rad, float force, float damage, float stun, float deafen, OnlinePhysicalObject killTagHolder, float killTagHolderDmgFactor, float minStun, float backgroundNoise, float hitfac, int hitChunk)
         {
-            objectHit?.apo.realizedObject.HitByExplosion(hitfac,null,0);
+            var source = (sourceObject?.apo.realizedObject);
+            // can null ref but wont kill game
+            var creature = (killTagHolder.apo as AbstractCreature).realizedCreature;
+            var room = (objectHit.apo.Room.realizedRoom);
+            var explosion = new Explosion(room, source, pos, lifeTime, rad, force, damage, stun, deafen, creature, killTagHolderDmgFactor, minStun, backgroundNoise);
+
+            objectHit.apo.realizedObject.HitByExplosion(hitfac, explosion, hitChunk);
```

Comment:
Make explosion null to prevent lag from multi-explosion events. Should maintain stability since we are giving it the room data now.